daal4py/sklearn/neighbors/_base.py

```diff
@@ -0,0 +1,416 @@
+        and single_output and fptype is not None and not sp.issparse(X):
+            logging.info("sklearn.neighbors.NeighborsBase._fit: " + method_uses_daal)
+            daal4py_fit(self, X, fptype)
+            result = self
+        else:
+            logging.info("sklearn.neighbors.NeighborsBase._fit: " + method_uses_sklearn)
+            if SKLEARN_24:
+                result = super(NeighborsBase, self)._fit(X, y)
+            else:
+                result = super(NeighborsBase, self)._fit(X)
+
+        return result
+
+
+class KNeighborsMixin(BaseKNeighborsMixin):
+    def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
+        daal_model = getattr(self, '_daal_model', None)
+        x = self._fit_X if X is None else X
+        try:
+            fptype = getFPType(x)
+        except ValueError:
+            fptype = None
+
+        if daal_check_version(((2020,'P', 3),(2021,'B', 110))) and not (daal_model is None) \
+        and fptype is not None and not sp.issparse(X):
+            logging.info("sklearn.neighbors.KNeighborsMixin.kneighbors: " + method_uses_daal)
+            result = daal4py_kneighbors(self, X, n_neighbors, return_distance)
+        else:
+            logging.info("sklearn.neighbors.KNeighborsMixin.kneighbors: " + method_uses_sklearn)
+            if getattr(self, '_tree', 0) is None and self._fit_method == 'kd_tree':
+                raise ValueError('oneDAL does not build sklearn.neighbors._kd_tree.KDTree')
+            if not (daal_model is None):
+                if SKLEARN_24:
+                    BaseNeighborsBase._fit(self, self._fit_X, self._y)
+                else:
+                    BaseNeighborsBase._fit(self, self._fit_X)
+            result = super(KNeighborsMixin, self).kneighbors(X, n_neighbors, return_distance)
+
+        return result
+
+
+class RadiusNeighborsMixin(BaseRadiusNeighborsMixin):
+    def radius_neighbors(self, X=None, radius=None, return_distance=True,
+                         sort_results=False):
+        if getattr(self, '_tree', 0) is None and self._fit_method == 'kd_tree':
+            raise ValueError('oneDAL does not build sklearn.neighbors._kd_tree.KDTree')
+        return BaseRadiusNeighborsMixin.radius_neighbors(self, X, radius, return_distance, sort_results)
+
+
+def daal4py_classifier_predict(estimator, X, base_predict):
+    X = check_array(X, accept_sparse='csr')
+    daal_model = getattr(estimator, '_daal_model', None)
+
+    n_features = getattr(estimator, 'n_features_in_', None)
+    shape = getattr(X, 'shape', None)
+    if n_features and shape and len(shape) > 1 and shape[1] != n_features:
+        raise ValueError('Input data shape {} is inconsistent with the trained model'.format(X.shape))
+
+    try:
+        fptype = getFPType(X)
+    except ValueError:
+        fptype = None
+
+    if daal_check_version(((2020,'P', 3),(2021,'B', 110))) and not (daal_model is None) \
+    and fptype is not None and not sp.issparse(X):
```

Comment:
I think should be enough daal_model is None

Reply:
Fixed